File: jour_27/jour27.py
#creation de l'enregistrement etudiant
class Etudiant : 
    def __init__(self,nom,age,note) :
        self.nom = nom
        self.age = age 
        self.note = note 
tab_etu = []
while True :
    try :
        nombre_etu = int(input("Entrez le nombre d'etudiants : "))
        if nombre_etu <= 0 : 
            print("Veuillez entrer un nombre positif\n")
        else : 
            break
    except ValueError : 
        print("saisie invalide\n")
#saisie des data des etudiants
for i in range(nombre_etu) : 
    print(f"\n======= Etudiant N {i + 1} =======")
    nom = input("Nom : ")
    age = int(input("Age : "))
    note = float(input("Note : "))
    tab_etu.append(Etudiant(nom,age,note))

File: jour_27/jour27.pas
program GestionEtudiants;

type 
    Etudiants = record  
        nom : string;
        age : integer;
        note : real;
    end;

var
    nombre_etu, i : integer;
    tab_etu : array[1..100] of Etudiants; 

begin 
    repeat 
        write('Entrez le nombre d etudiants : ');
        readln(nombre_etu);
        if nombre_etu <= 0 then
        begin   
            writeln('Entrez un nombre positif');
        end;
    until(nombre_etu > 0);

    { Saisie des data des etudiants }
    for i := 1 to nombre_etu do
    begin
        writeln('======= Etudiant N ', i, ' =======');
        write('Nom : '); readln(tab_etu[i].nom);
        write('Age : '); readln(tab_etu[i].age);
        write('Note : '); readln(tab_etu[i].note);
    end;
end.